runtime/src/system_instruction_processor.rs

```diff
@@ -621,6 +771,31 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_assign_from_nonce_account_fail() {
```

Comment:
glad you're checking, but I think that verify_instruction also guards against this...

Reply:
Mind pointing to where you're talking about? `git grep` is failing me

Reply:
ah, was renamed to `verify_account_changes()`

Reply:
Looks like it.  The `!is_zeroed()` check gets us.  So I guess what we're buying here is a bit of accuracy for the return code, `InvalidArgument` vs. `ModifiedProgramId`

Reply:
`verify_account_changes()` can be really explicit, have a ton of errors, all part of the base set.  the rules apply to *all* programs, not just system.

legal under the current rules: the stake program can change `account.owner` if it first zeros the state memory.  that's pretty cool and correct, right?